src/system_info/system_info_network_mobile.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2013 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SYSTEM_INFO_SYSTEM_INFO_NETWORK_MOBILE_H_
+#define SYSTEM_INFO_SYSTEM_INFO_NETWORK_MOBILE_H_
+
+#include <vconf.h>
+
+#include <string>
+
+#include "common/picojson.h"
+#include "common/utils.h"
+#include "system_info/system_info_instance.h"
+#include "system_info/system_info_utils.h"
+
+enum SystemInfoNetworkType {
+  SYSTEM_INFO_NETWORK_NONE = 0,
+  SYSTEM_INFO_NETWORK_2G,
+  SYSTEM_INFO_NETWORK_2_5G,
+  SYSTEM_INFO_NETWORK_3G,
+  SYSTEM_INFO_NETWORK_4G,
+  SYSTEM_INFO_NETWORK_WIFI,
+  SYSTEM_INFO_NETWORK_ETHERNET,
+  SYSTEM_INFO_NETWORK_UNKNOWN
+};
+
+class SysInfoNetwork : public SysInfoObject {
+ public:
+  static SysInfoObject& GetInstance() {
+    static SysInfoNetwork instance;
+    return instance;
+  }
+  ~SysInfoNetwork();
+  void Get(picojson::value& error, picojson::value& data);
+  void StartListening();
+  void StopListening();
+
+  static const std::string name_;
```

Comment:
The interface code is duplicated in all configuratoin(_desktop, _ivi, _mobile), could you design a pure class with those common interface and all configuration .h/.cc inherit on it. @qjia7 please guide @liyingzh for that.